internal/controller/endpoint/integrations/kubernetes/visibility/strategy.go

```diff
@@ -62,6 +63,14 @@ func (s *PublicVisibilityStrategy) IsHTTPRouteRequired(epCtx *dataplane.Endpoint
 		epCtx.Endpoint.Spec.NetworkVisibilities.Public.Enable
 }
 
+func (s *PublicVisibilityStrategy) IsHTTPRouteFilterRequired(epCtx *dataplane.EndpointContext) bool {
+	if epCtx.Component.Spec.Type == choreov1.ComponentTypeWebApplication || epCtx.Endpoint.Spec.NetworkVisibilities == nil {
+		return false
+	}
+	return epCtx.Endpoint.Spec.NetworkVisibilities.Public != nil &&
+		epCtx.Endpoint.Spec.NetworkVisibilities.Public.Enable
+}
```

Comment:
Isn't the filter always required since we're no longer rewriting the paths using prefix-rewrite?